src/AutoRest.CSharp/Mgmt/AutoRest/MgmtOutputLibrary.cs

```diff
@@ -96,24 +90,32 @@ internal class MgmtOutputLibrary : OutputLibrary
         /// </summary>
         public HashSet<TypeProvider> PropertyBagModels { get; }
 
-        public MgmtOutputLibrary()
+        public TypeFactory TypeFactory { get; }
+
+        public MgmtOutputLibrary(CodeModel codeModel, SchemaUsageProvider schemaUsages)
         {
+            TypeFactory = new TypeFactory(this);
+
             ApplyGlobalConfigurations();
             CodeModelTransformer.Transform();
 
             // these dictionaries are initialized right now and they would not change later
-            RawRequestPathToOperationSets = CategorizeOperationGroups();
+            RawRequestPathToOperationSets = CategorizeOperationGroups(codeModel);
             ResourceDataSchemaNameToOperationSets = DecorateOperationSets();
+            _schemaToInputEnumMap = new CodeModelConverter(codeModel, schemaUsages).CreateEnums();
+
+            AllSchemaMap = InitializeModels(codeModel);
+
+            var codeModelConverter = new CodeModelConverter(codeModel, schemaUsages);
+            _input = codeModelConverter.CreateNamespace();
+            _inputOperationToOperation = codeModelConverter.GetCurrentInputOperationToOperationMap();
```

Comment:
So we convert the code model to input types here, but similar to the legacy data plane, we are only using them to generate the operations, models keep the previous behavior to use the code model schemas to generate. Is this correct?

Reply:
In HLC, models are generated from input types. In MGP, that is an upcoming work:
https://github.com/Azure/autorest.csharp/issues/3770